applications/StructuralMechanicsApplication/python_scripts/adaptative_structural_mechanics_analysis.py

```diff
@@ -0,0 +1,218 @@
+try:
+    import KratosMultiphysics.ExternalSolversApplication
+    KratosMultiphysics.Logger.PrintInfo("ExternalSolversApplication", "succesfully imported")
+except ImportError:
+    KratosMultiphysics.Logger.PrintInfo("ExternalSolversApplication", "not imported")
+try:
+    import KratosMultiphysics.EigenSolversApplication
+    KratosMultiphysics.Logger.PrintInfo("EigenSolversApplication", "succesfully imported")
+except ImportError:
+    KratosMultiphysics.Logger.PrintInfo("EigenSolversApplication", "not imported")
+try:
+    import KratosMultiphysics.MeshingApplication as MA
+    KratosMultiphysics.Logger.PrintInfo("MeshingApplication", "succesfully imported")
+except ImportError:
+    KratosMultiphysics.Logger.PrintInfo("MeshingApplication", "not imported")
+
+# Other imports
+import sys
+
+# Import the base structural analysis
+from structural_mechanics_analysis import StructuralMechanicsAnalysis as BaseClass
+
+class AdaptativeStructuralMechanicsAnalysis(BaseClass):
+    """
+    This class is the main-script of the StructuralMechanicsApplication when using adaptative remeshing put in a class
+
+    It can be imported and used as "black-box"
+    """
+    def __init__(self, model, project_parameters):
+
+        # Construct the base analysis.
+        default_params = KM.Parameters("""
+        {
+            "max_iteration" : 1,
+            "analysis_type" : "linear"
+        }
+        """)
+        if (project_parameters["solver_settings"].Has("max_iteration") is True):
+            self.non_linear_iterations = project_parameters["solver_settings"]["max_iteration"].GetInt()
+        else:
+            self.non_linear_iterations = 10
+            project_parameters["solver_settings"].AddValue("max_iteration", default_params["max_iteration"])
+        if (project_parameters["solver_settings"].Has("analysis_type") is True):
+            project_parameters["solver_settings"]["analysis_type"].SetString("linear")
+        else:
+            project_parameters["solver_settings"].AddValue("analysis_type", default_params["analysis_type"])
+        if (project_parameters.Has("recursive_remeshing_process") is True):
+            self.process_remesh = True
+        else:
+            self.process_remesh = False
+        super(AdaptativeStructuralMechanicsAnalysis, self).__init__(model, project_parameters)
+
+    def Initialize(self):
+        """ Initializing the Analysis """
+        super(AdaptativeStructuralMechanicsAnalysis, self).Initialize()
+        if (self.process_remesh is False):
+            convergence_criteria = self._GetSolver().get_convergence_criterion()
+            convergence_criteria.Initialize(self._GetSolver().GetComputingModelPart())
+        # Ensuring to have conditions on the BC before remesh
+        computing_model_part = self._GetSolver().GetComputingModelPart()
+        # We need to detect the conditions in the boundary conditions
+        if (self.project_parameters.Has("constraints_process_list") is True):
```

Comment:
this is deprecated, you should also look into ["processes"]["constraints_process_list"]